aodndata/soop/soop_co2_nrt_nc_generator.py

```diff
@@ -0,0 +1,355 @@
+"""
+# class to generate NetCDF file storing realtime SOOP-CO2 data.
+# Script processes data from 2 vessels: RV Aurora Australis and RV Investigator
+# -Exctract varaiable relevant to CO2 measurements and processing from input text file.
+# -Map vessel specific variable to set of common output variables
+# Mapping as follows:
+#       Input Variable                  NetCDF Variable
+#   Aurora      Investigator
+#       PcDate+PcTime                        TIME
+#       GpsShipLatitude                     LATITUDE
+#       GpsShipLongitude                    LONGITUDE
+#           Type                            TYPE
+#           EquTemp                         TEQ_raw
+#           CO2StdValue                     CO2_STD_Value
+#           CO2um_m                         xCO2_PPM_raw
+#           H2Omm_m                         xH2O_PPT_raw
+#       DryBoxDruckPress                    Press_Licor_raw
+#           EquPress                        Diff_Press_Equ_raw
+#           EquH2OFlow                      H2O_flow_raw
+#           LicorFlow                       Licor_flow_raw
+#           AtmSeaLevelPress                ATMP_raw
+#           MetTrueWindSpKts                WSPD_raw
+#           MetTrueWindDir                  WDIR_raw
+#           IntakeShipTemp                  TEMP_raw
+# TsgSbe45Salinity  TsgShipSalinity          PSAL_raw
+# TsgSbe45Temp      TsgShipTemp             TEMP_Tsg_raw
+# SBE45Flow         TsgShipFlow             Tsg_flow_raw
+#    -             LabMainSwFlow            LabMain_sw_flow_raw
+"""
+
+from aodncore.pipeline.exceptions import InvalidFileContentError
+import os
+from datetime import datetime
+import pandas as pd
+import numpy as np
+from netCDF4 import Dataset, stringtochar
+import re
+import collections
+from tempfile import mkdtemp
+
+from ship_callsign import ship_callsign_list, ship_callsign
+
+VALID_PROJECT = ['IMOS', 'FutureReefMap', 'SOOP-CO2_RT']
+INPUT_RT_PARAMETERS = {'Type', 'PcDate', 'PcTime', 'GpsShipLatitude',
+                       'GpsShipLongitude', 'EquTemp', 'CO2StdValue',
+                       'CO2um_m', 'H2Omm_m', 'DryBoxDruckPress', 'EquPress',
+                       'EquH2OFlow', 'LicorFlow', 'IntakeShipTemp', 'MetTrueWindSpKts',
+                       'MetTrueWindDir', 'AtmSeaLevelPress'}
+AA_SPECIFIC_INPUT_PARAMS = {'SBE45Flow', 'TsgSbe45Temp',
+                            'TsgSbe45Salinity'}
+IN_SPECIFIC_INPUT_PARAMS = {'TsgShipTemp',
+                            'TsgShipSalinity', 'TsgShipFlow', 'LabMainSwFlow'}
+
+VESSEL = {
+    'AA': 'VNAA',
+    'IN': 'VLMJ'}
+
+
+def process_co2_rt(self):
+    """
+    Read in data from co2 realtime file and produce a netcdf file
+    """
+    # Parse data into dataframe
+    (dataf, platform_code) = read_realtime_file(self)
+    # format data
+    (dtime, time) = get_time_formatted(dataf)
+    # generate nc file name
+    netcdf_filename = create_netcdf_filename(platform_code, dtime)
+    tmpdir = mkdtemp()
```

Comment:
This should be the products_dir from the handler instead of just a random temp dir. This helps make sure that the files all get created under the same temp dir for the handler.

You could update the function to accept as an argument, and then pass through "self.products_dir" from the handler when you call it, e.g.

```python
def process_co2_rt(realtime_file_path, temp_dir):
```